specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2023-01-01-preview/AlertRuleRecommendations.json

```diff
@@ -0,0 +1,251 @@
+          },
+          "List alert rule recommendations for Monitoring accounts at resource level": {
+            "$ref": "./examples/AlertRuleRecommendations_GetByResource_MAC.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.AlertsManagement/alertRuleRecommendations": {
+      "get": {
+        "tags": [
+          "alertRuleRecommendations"
+        ],
+        "description": "Retrieve alert rule recommendations for a target type.",
+        "operationId": "AlertRuleRecommendations_ListByTargetType",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/TargetTypeParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successful request for a list of alert rule recommendations",
+            "schema": {
+              "$ref": "#/definitions/AlertRuleRecommendationsListResponse"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "List alert rule recommendations for virtual machines at subscription level": {
+            "$ref": "./examples/AlertRuleRecommendations_GetBySubscription_VM.json"
+          },
+          "List alert rule recommendations for Monitoring accounts at subscription level": {
+            "$ref": "./examples/AlertRuleRecommendations_GetBySubscription_MAC.json"
+          }
+        }
+      }
+    }
+  },
+  "parameters": {
+    "ResourceUriParameter": {
+      "name": "resourceUri",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "The identifier of the resource.",
+      "x-ms-parameter-location": "method",
+      "x-ms-skip-url-encoding": true
+    },
+    "TargetTypeParameter": {
+      "name": "targetType",
+      "in": "query",
+      "required": true,
+      "type": "string",
+      "description": "The recommendations target type.",
+      "x-ms-parameter-location": "client"
+    }
+  },
+  "definitions": {
+    "AlertRuleRecommendationsListResponse": {
+      "type": "object",
+      "description": "List of alert rule recommendations.",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/AlertRuleRecommendation"
+          },
+          "description": "the values for the alert rule recommendations."
+        }
+      }
+    },
+    "AlertRuleRecommendation": {
```

Comment:
>AlertRuleRecommendation [](http://example.com/codeflow?start=5&length=23)

AlertRuleRecommendation should be a valid ARM Resource, means, it should be either Tracked Resource or Proxy Resource, using the common-types. Every ARM Resource needs to have id, name, type and properties as required properties as well.

Reply:
That's an extension(proxyOnly) resource. 
Do I still have to return Id, name, type and properties? 
I was looking at other example (microsoft.insights/metricDefinitions) - and it doesn't seem like it has all that.

Reply:
Add required properties